dds/DCPS/ReceivedDataElementList.cpp

```diff
@@ -131,8 +147,138 @@ OpenDDS::DCPS::ReceivedDataElementList::remove(
 }
 
 bool
-OpenDDS::DCPS::ReceivedDataElementList::remove(ReceivedDataElement *data_sample)
+OpenDDS::DCPS::ReceivedDataElementList::remove(ReceivedDataElement* data_sample)
 {
   IdentityFilter match(data_sample);
   return remove(match, false); // short-circuit evaluation
 }
+
+bool
+OpenDDS::DCPS::ReceivedDataElementList::has_zero_copies() const
+{
+  for (ReceivedDataElement* item = head_; item != 0; item = item->next_data_sample_) {
+#ifdef ACE_HAS_CPP11
+    if (item->zero_copy_cnt_) {
+#else
+    if (item->zero_copy_cnt_.value()) {
+#endif
+      return true;
+    }
+  }
+  return false;
+}
+
+bool
+OpenDDS::DCPS::ReceivedDataElementList::matches(CORBA::ULong sample_states) const
+{
+  return sample_states_ & sample_states;
+}
+
+OpenDDS::DCPS::ReceivedDataElement*
+OpenDDS::DCPS::ReceivedDataElementList::get_next_match(CORBA::ULong sample_states, ReceivedDataElement* prev)
+{
+  OPENDDS_ASSERT(sanity_check(prev));
+  if (prev == tail_) {
+    return NULL;
```

Comment:
```suggestion
    return 0;
```
a few more of these below